### ISSUE TITLE
RdLogManager event managing segfaults

### ISSUE THREAD

**Issue by alexolivan** [ElvishArtisan/rivendell#711]
Opening new, separate issue.

My builds are stuck on 08/17 commits, so not sure if recent commits could have render this issue invalid.
Anyhow, the issue is IMHO frontend/qt5 related: Apparently the pre/post import cart/macro code is broken in a way that, as soon as you start creating events and adding macros/carts, they seem to populate some kind of iterative data structures that suffer an out-of-bound exception.

```
operador@debian:~$ rdlogmanager
QList::insert(): Index out of range.
QList::insert(): Index out of range.
QList::insert(): Index out of range.
QList::insert(): Index out of range.
QList::insert(): Index out of range.
Segmentation fault
```

_Originally posted by @alexolivan in https://github.com/ElvishArtisan/rivendell/issues/706#issuecomment-903166689_

**Comment by ElvishArtisan** [comment]
Confirmed here.

**Comment by ElvishArtisan** [comment]
Fixed in fc61172. Please test!

**Comment by alexolivan** [comment]
Confirmed fixed.
Closing. Thank you!
